internal/maintenance/periodic_job_enqueuer_test.go

```diff
@@ -264,39 +259,226 @@ func TestPeriodicJobEnqueuer(t *testing.T) {
+		})
+
+		svc.TestSignals.InsertedJobs.WaitOrTimeout()
+		requireNJobs(t, bundle.exec, "periodic_job_500ms", 0)
+		requireNJobs(t, bundle.exec, "periodic_job_500ms_start", 1)
+
+		svc.Remove(handles[1])
+
+		// Each is one because the second job was removed before it was worked
+		// again.
+		svc.TestSignals.InsertedJobs.WaitOrTimeout()
+		requireNJobs(t, bundle.exec, "periodic_job_500ms", 1)
+		requireNJobs(t, bundle.exec, "periodic_job_500ms_start", 1)
+	})
+
+	t.Run("RemoveManyAfterStart", func(t *testing.T) {
+		t.Parallel()
+
+		svc, bundle := setup(t)
+
+		startService(t, svc)
+
+		handles := svc.AddMany([]*PeriodicJob{
+			{ScheduleFunc: periodicIntervalSchedule(500 * time.Millisecond), ConstructorFunc: jobConstructorFunc("periodic_job_500ms", false)},
+			{ScheduleFunc: periodicIntervalSchedule(500 * time.Millisecond), ConstructorFunc: jobConstructorFunc("periodic_job_500ms_other", false)},
+			{ScheduleFunc: periodicIntervalSchedule(500 * time.Millisecond), ConstructorFunc: jobConstructorFunc("periodic_job_500ms_start", false), RunOnStart: true},
+		})
+
+		svc.TestSignals.InsertedJobs.WaitOrTimeout()
+		requireNJobs(t, bundle.exec, "periodic_job_500ms", 0)
+		requireNJobs(t, bundle.exec, "periodic_job_500ms_other", 0)
+		requireNJobs(t, bundle.exec, "periodic_job_500ms_start", 1)
+
+		svc.RemoveMany([]rivertype.PeriodicJobHandle{handles[1], handles[2]})
+
+		// Each is one because the second job was removed before it was worked
+		// again.
+		svc.TestSignals.InsertedJobs.WaitOrTimeout()
+		requireNJobs(t, bundle.exec, "periodic_job_500ms", 1)
+		requireNJobs(t, bundle.exec, "periodic_job_500ms_other", 0)
+		requireNJobs(t, bundle.exec, "periodic_job_500ms_start", 1)
+	})
+
+	// To suss out any race conditions in the add/remove/clear/run loop code,
+	// and interactions between them.
+	t.Run("AddRemoveStress", func(t *testing.T) {
+		t.Parallel()
+
+		svc, _ := setup(t)
+
+		var wg sync.WaitGroup
+
+		randomSleep := func() time.Duration {
+			return time.Duration(randutil.IntBetween(svc.Rand, 1, 5)) * time.Millisecond
+		}
+
+		for i := 0; i < 10; i++ {
+			wg.Add(1)
+
+			jobBaseName := fmt.Sprintf("periodic_job_1ms_%02d", i)
+
+			go func() {
+				defer wg.Done()
+
+				for j := 0; j < 50; j++ {
+					handle := svc.Add(&PeriodicJob{ScheduleFunc: periodicIntervalSchedule(time.Millisecond), ConstructorFunc: jobConstructorFunc(jobBaseName, false)})
+					randomSleep()
+
+					svc.Add(&PeriodicJob{ScheduleFunc: periodicIntervalSchedule(time.Millisecond), ConstructorFunc: jobConstructorFunc(jobBaseName+"_second", false)})
+					randomSleep()
+
+					svc.Remove(handle)
+					randomSleep()
+
+					svc.Clear()
+					randomSleep()
+				}
+			}()
+		}
+
+		wg.Wait()
+	})
+
 	t.Run("NoJobsConfigured", func(t *testing.T) {
 		t.Parallel()
 
 		svc, _ := setup(t)
 
-		svc.periodicJobs = []*PeriodicJob{}
+		// TODO: How about just don't configure the jobs in the first place.
```

Comment:
missed TODO?

Reply:
Yep, oops. I ended up changing the setup function to not add default jobs, but forgot to fix this.